### DISCUSSION TITLE
Question: Passive Mode

### DISCUSSION THREAD

**Discussion by pagner** [PiranhaCMS/piranha.core#2102] [Q&A] [answered]
First of all, I'd like to say what a nice job you've done! I installed the default demo and I am inpressed with the quality of work. Good stuff.

I read some posts on Stack Overflow that you replied to about passive mode.

I want to add Piranha to an existing site and only use it for content management. I do not want it handling the routing at all. I already have my routing set up for localization using a sub-directory structure, i.e. mysite.com, mysite.com/es-es/, mysite.com/es-es/mypage.

 Does passive mode still exist? Will I be able to accomplish the aforementioned?

Thanks

**Comment by pagner** [comment]
Had to do a search for headless. 

From what I have read, I can use IApi to fetch data from the repository, but it appears I will be responsible for deseralizing on the front end? Are there any packages for the front end that will do that for me, so I can just loop through the blocks like in your demo example? Thanks.

**Comment by tidyui** [comment, marked as answer]
Hi there! In startup you can simply disable all routing features to get it to run in "passive mode".

~~~csharp
services.AddPiranha(options =>
{
    options.UseCms(o =>
    {
        o.UseAliasRouting = false;
        o.UseArchiveRouting = false;
        o.UsePageRouting = false;
        o.UsePostRouting = false;
        o.UseSitemapRouting = false;
        o.UseSiteRouting = false;
        o.UseStartpageRouting = false;
    });
});
~~~
